mypy/semanal.py

```diff
@@ -1794,11 +1603,11 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
             for lvalue in s.lvalues:
                 self.store_declared_types(lvalue, s.type)
         self.check_and_set_up_type_alias(s)
-        self.process_newtype_declaration(s)
+        self.newtype_analyzer.process_newtype_declaration(s)
```

Comment:
Rather a comment for myself: `NewType` and `TypeVar` not requiring `self.is_func_scope()` are by interesting coincidence(?) behave in weird way with forward references in fine grained, see e.g. https://github.com/python/mypy/issues/4615